### PR TITLE
issue-590, ingestNodes for OpenSearch in API v2

### PR DESCRIPTION
**Pull request by OleksiienkoMykyta**

### CODE REVIEW

File: apis/clusters/v1beta1/opensearch_types.go [hunk 1]
```diff
@@ -55,7 +55,8 @@ type OpenSearchSpec struct {
 	BundledUseOnly           bool                    `json:"bundledUseOnly,omitempty"`
 	UserRefs                 []*UserReference        `json:"userRefs,omitempty"`
 	//+kubuilder:validation:MaxItems:=1
-	ResizeSettings []*ResizeSettings `json:"resizeSettings,omitempty"`
+	ResizeSettings       []*ResizeSettings       `json:"resizeSettings,omitempty"`
+	OpenSearchIngestNode []*OpenSearchIngestNode `json:"ingestNodes,omitempty"`
```
Comment (testisnullus) [inline apis/clusters/v1beta1/opensearch_types.go:59]: OpenSearchIngestNode -> OpenSearchIngestNodes

**Review by testisnullus — CHANGES_REQUESTED** [review]

File: apis/clusters/v1beta1/opensearch_types.go (hunk 1 above)
Comment (OleksiienkoMykyta) [reply apis/clusters/v1beta1/opensearch_types.go]: fixed

**Review by tengu-alt — APPROVED** [review]

File: apis/clusters/v1beta1/opensearch_types.go [hunk 2]
```diff
@@ -56,6 +56,8 @@ type OpenSearchSpec struct {
 	UserRefs                 []*UserReference        `json:"userRefs,omitempty"`
 	//+kubuilder:validation:MaxItems:=1
 	ResizeSettings []*ResizeSettings `json:"resizeSettings,omitempty"`
+	//+kubuilder:validation:MaxItems:=1
+	OpenSearchIngestNodes []*OpenSearchIngestNodes `json:"ingestNodes,omitempty"`
```
Comment (ribaraka) [inline apis/clusters/v1beta1/opensearch_types.go:60]: OpenSearch- prefix in field name is unnecessary.

Comment (testisnullus) [inline apis/clusters/v1beta1/opensearch_types.go:60]: []*OpenSearchIngestNodes -> []*OpenSearchIngestNode

File: controllers/clusterresources/nodereload_controller.go [hunk 3]
```diff
@@ -185,10 +185,10 @@ func (r *NodeReloadReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 
 	l.Info("The node has been successfully reloaded",
-		"Node ID", nrs.Status.NodeInProgress.ID,
+		"Node ID", nodeReloadStatus.NodeID,
 	)
 	r.EventRecorder.Eventf(nrs, models.Normal, models.UpdatedEvent,
-		"Node %s has been successfully reloaded", nrs.Status.NodeInProgress.ID,
+		"Node %s has been successfully reloaded", nodeReloadStatus.NodeID,
```
Comment (testisnullus) [inline controllers/clusterresources/nodereload_controller.go:191]: Revert these changes please

File: apis/clusters/v1beta1/opensearch_types.go [hunk 4]
```diff
@@ -211,6 +219,17 @@ func (oss *OpenSearchSpec) dataNodesToInstAPI() (iDataNodes []*models.OpenSearch
 	return
 }
 
+func (oss *OpenSearchSpec) ingestNodesToInstAPI() (iIngestNodes []*models.OpenSearchIngestNodes) {
+	for _, ingestNodes := range oss.OpenSearchIngestNodes {
```
Comment (testisnullus) [inline apis/clusters/v1beta1/opensearch_types.go:223]: ingestNodes -> ingestNode

**Review by testisnullus — CHANGES_REQUESTED** [review]

**Review by ribaraka — APPROVED** [review]

File: pkg/models/opensearch_apiv2.go [hunk 5]
```diff
@@ -18,29 +18,35 @@ package models
...
+	PrivateNetworkCluster    bool                     `json:"privateNetworkCluster"`
+	Name                     string                   `json:"name"`
+	BundledUseOnly           bool                     `json:"bundledUseOnly"`
+	ClusterManagerNodes      []*ClusterManagerNodes   `json:"clusterManagerNodes"`
+	IndexManagementPlugin    bool                     `json:"indexManagementPlugin"`
+	SLATier                  string                   `json:"slaTier,omitempty"`
+	AlertingPlugin           bool                     `json:"alertingPlugin"`
+	ResizeSettings           []*ResizeSettings        `json:"resizeSettings,omitempty"`
+	Description              string                   `json:"description,omitempty"`
+	OpenSearchIngestNodes    []*OpenSearchIngestNodes `json:"ingestNodes,omitempty"`
```
Comment (worryg0d) [inline pkg/models/opensearch_apiv2.go:44]: `OpenSearch` prefix is not necessary because it's obvious that they are nodes of OpenSearch cluster

**Review by worryg0d — APPROVED** [review]
lgtm

File: controllers/clusterresources/nodereload_controller.go (hunk 3 above)
Comment (OleksiienkoMykyta) [reply controllers/clusterresources/nodereload_controller.go]: replaced with "nrs.Status"

File: apis/clusters/v1beta1/opensearch_types.go (hunk 4 above)
Comment (OleksiienkoMykyta) [reply apis/clusters/v1beta1/opensearch_types.go]: Fixed

File: apis/clusters/v1beta1/opensearch_types.go (hunk 2 above)
Comment (OleksiienkoMykyta) [reply apis/clusters/v1beta1/opensearch_types.go]: Fixed

File: pkg/models/opensearch_apiv2.go (hunk 5 above)
Comment (OleksiienkoMykyta) [reply pkg/models/opensearch_apiv2.go]: Fixed
